istioctl/pkg/util/formatting/formatter_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package formatting
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	. "istio.io/istio/galley/pkg/config/analysis/diag"
+)
+
+func TestFormatter_RenderWithColor(t *testing.T) {
+	g := NewWithT(t)
+	mt := NewMessageType(Error, "IST-0042", "Cheese type not found: %q")
+	m := NewMessage(mt, nil, "Feta")
+
+	g.Expect(render(m, true)).To(Equal(
+		colorPrefixes[Error] + "Error\033[0m [IST-0042] Cheese type not found: \"Feta\""))
+}
```

Comment:
Nit: ideally we can test all of the functionality of a class without touching non-exported pieces. Here if we want to check the color, I'd suggest checking it via Print(..., true) instead of poking directly at render() and colorPrefixes.

This serves two purposes.
1) It makes the tests more resistant to refactoring. Generally the public API doesn't change as much as the internals of a package, so if all of the tests go through exported methods there's less chance that changing the package means you also have to change a test.
2) It ensures that all of the code can actually get exercised with normal use. If there's not a way to test everything in the package without touching non-exported variables/functions, it's normally a sign that things might not be working as intended.

If you want to give it a try, I'd suggest changing the package to "formatting_test" instead of "formatting" so Go won't let you touch the internals. It's a small thing so I won't block acceptance on this.

Also, I think this entire file would also be a good candidate for table driven tests.